mnefun/_mnefun.py

```diff
@@ -2411,21 +2387,104 @@ def gen_html_report(p, subjects, structurals, run_indices=None,
             print('    For %s no reports generated for:\n        %s'
                   % (subj, missing))
         patterns = [t for t, b in zip(texts, bools) if b]
-        fnames = get_raw_fnames(p, subj, 'pca', False, False, run_indices[si])
-        if not fnames:
+        raw_fif_fn = get_raw_fnames(p, subj, 'raw', False, True,
+                                    run_indices[si])[0]
+        if not raw_fif_fn:
+            raise RuntimeError('Could not find raw files for '
+                               'reporting.')
+        pca_fif_fn = get_raw_fnames(p, subj, 'pca', False, False,
+                                    run_indices[si])[0]
+        if not pca_fif_fn:
             raise RuntimeError('Could not find any processed files for '
                                'reporting.')
-        info_fname = op.join(path, fnames[0])
-        struc = structurals[si]
-        report = Report(info_fname=info_fname, subject=struc,
+        report = Report(info_fname=raw_fif_fn, subject=struc,
                         baseline=_get_baseline(p))
-        report.parse_folder(data_path=path, mri_decim=10, n_jobs=p.n_jobs,
-                            pattern=patterns)
+        if p.n_jobs != 'cuda':
+            n_jobs = p.n_jobs
+        else:
+            n_jobs = 1
+        report.parse_folder(data_path=path, mri_decim=10, n_jobs=n_jobs,
+                            pattern=patterns, sort_sections=True)
+        # Add HP coils SNR
+        fig = plot_good_coils(read_raw_fif(raw_fif_fn, allow_maxshield='yes'))
+        report.add_figs_to_section(fig, captions='cHPI coils',
+                                   section='cHPI')
+        # Add head movement traces
+        pos = _headpos(p, raw_fif_fn)
+        fig = plot_head_positions(pos=pos, show=False)
```

Comment:
We should modify this figure to show the median or time-weighted head position values using `ax.hlines`. I can do this in a follow-up PR, though, after this one is merged.